spec/publisher/unpublishing_with_redirect_spec.rb

```diff
@@ -0,0 +1,89 @@
+feature "Unpublishing with redirect on Publisher", publisher: true do
+  include PublisherHelpers
+
+  let(:title) { title_with_timestamp }
+  let(:slug) { slug_with_timestamp }
+  let(:redirect_url) { "https://www.gov.uk/help" }
+
+  scenario "Unpublishing an Artefact" do
+    given_there_is_a_published_artefact_with_subpages
+    when_i_remove_the_published_artefact_with_a_redirect_to_help
+    then_visiting_the_artefact_redirects_to_help
+    and_visiting_a_subpage_redirects_to_help_with_slug
+  end
+
+  private
+
+  def given_there_is_a_published_artefact_with_subpages
+    create_guide
+    fill_in_part_form
+    @subpage_slug = fill_in_part_form
+    @published_url = publish_artefact
+  end
+
+  def create_guide
+    visit_publisher("/artefacts/new")
+    fill_in_new_artefact_form(title: title, slug: slug, format: "Guide")
+    click_button "Save and go to item"
+
+    expect(page).to have_text(title)
+  end
+
+  def when_i_remove_the_published_artefact_with_a_redirect_to_help
+    switch_to_tab "Unpublish"
+
+    fill_in "redirect_url", with: redirect_url
+
+    page.accept_confirm do
+      click_button "Unpublish"
+    end
+
+    expect(page).to have_content("Content unpublished")
+  end
+
+  def then_visiting_the_artefact_redirects_to_help
+    reload_url_until_status_code(@published_url, 301, keep_retrying_while: [200])
+
+    visit(@published_url)
+
+    expect_rendering_application("frontend")
+    expect(page).to have_content("Help")
```

Comment:
I wonder if these are needed - as we only need to determine it's been a redirect rather than what's on the result of the redirect